packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -45,20 +48,29 @@ export abstract class TerminalWidget extends BaseWidget {
 
     abstract scrollToTop(): void;
 
+    abstract scrollToBottom(): void;
+
     abstract scrollPageUp(): void;
 
     abstract scrollPageDown(): void;
 
+    abstract resetTerminal(): void;
     /**
      * Event which fires when terminal did closed. Event value contains closed terminal widget definition.
      */
     abstract onTerminalDidClose: Event<TerminalWidget>;
 
+    abstract onTerminalDidDispose: Event<void>;
+
+    abstract onDidReconnectTerminalProcess: Event<TerminalWidget>;
```

Comment:
why don't have `onDidConnect` which fires whenever a connection is established? Is it an event that `isInteracitve` flag changed?

Reply:
yep. my `onDidReconnectTerminalProcess` does the job of `onDidConnect` in your comment. I should rename it.